qiskit/primitives/backend_estimator_v2.py

```diff
@@ -162,7 +162,11 @@ def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
         pub_dict = defaultdict(list)
         # consolidate pubs with the same number of shots
         for i, pub in enumerate(pubs):
-            shots = int(math.ceil(1.0 / pub.precision**2))
+            coeff_sq_sum = 0
+            for obs in pub.observables.ravel():
+                # each element of pub.observables.ravel() may be a multi-term observable
+                coeff_sq_sum = max(coeff_sq_sum, sum(coeff**2 for coeff in obs.values()))
+            shots = int(math.ceil(coeff_sq_sum / pub.precision**2))
```

Comment:
If all coefficients are 0, shots is set 0 and it results in an error. You need to set at least 1 shot.
Could you add the following case to the unit test?
```suggestion
            shots = int(math.ceil(coeff_sq_sum / pub.precision**2))
            shots = max(1, shots)
```

The following script causes an error only for this PR.
```python
from qiskit import QuantumCircuit
from qiskit.primitives import BackendEstimatorV2, StatevectorEstimator
from qiskit_aer import AerSimulator
from qiskit.quantum_info import SparsePauliOp

backend = AerSimulator()
estimator = BackendEstimatorV2(backend=backend)

qc = QuantumCircuit(1)
qc.h(0)
op = SparsePauliOp.from_list([("X", 0), ("Y", 0)])

result = estimator.run([(qc, op)]).result()
print(result[0].data.evs)
```

this PR
```
qiskit.exceptions.QiskitError: 'No counts for experiment "0"'
```

main
```
0.0
```

Reply:
@t-imamichi I see, thanks. Could I use one line?
`shots = max(1, int(math.ceil(coeff_sq_sum / pub.precision**2)))`

Reply:
Sure